sdks/java/io/debezium/src/main/java/org/apache/beam/io/debezium/KafkaSourceConsumerFn.java

```diff
@@ -202,6 +209,63 @@ private static Instant ensureTimestampWithinBounds(Instant timestamp) {
     return timestamp;
   }
 
+  private static final Cache<Map<String, String>, SourceTask> CONNECTOR_CACHE =
+      CacheBuilder.newBuilder()
+          .expireAfterAccess(java.time.Duration.ofSeconds(60))
+          .removalListener(
+              new RemovalListener<Map<String, String>, SourceTask>() {
+                @Override
+                public void onRemoval(
+                    RemovalNotification<Map<String, String>, SourceTask> removalNotification) {
+                  LOG.debug(
+                      "Task for key [[{}]] is being removed. Cause: {}",
+                      removalNotification.getKey(),
+                      removalNotification.getCause());
+                  removalNotification.getValue().stop();
+                }
+              })
+          .maximumSize(10)
+          .build();
+
+  SourceTask getDebeziumSourceTask(
+      Map<String, String> element, RestrictionTracker<OffsetHolder, Map<String, Object>> tracker) {
+    final KafkaSourceConsumerFn<T> dofnInstance = this;
+    try {
```

Comment:
If we have a value in the cache, do we need to update the consumer offset on what is being returned (we currently only do it on the first creation)?

Reply:
we don't need to. The task should always have an up-to-date offset with the latest offset. However I want to test it in the next PR (add some sort of asserttion on the latest initialized context)

Reply:
pleaseee I have an item to track this and verify it will work : D